projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -46,17 +46,18 @@ export class IgxExcelStyleSearchComponent {
 
     public onCheckboxChange(eventArgs: IChangeCheckboxEventArgs) {
         const selectAll = this.column.grid.resourceStrings.igx_grid_excel_select_all;
-        if (eventArgs.checkbox.value.value === selectAll) {
```

Comment:
This code will work fine without this modifications. But if you want to do them I suggest several modification. See my comments bellow.

Reply:
The check for selected index is necessary, since if there is another checkbox with 'Select All' value it behaves like the main select all.